digitalocean/spaces/resource_spaces_bucket_cors_configuration_test.go

```diff
@@ -0,0 +1,328 @@
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "cors_rule.*", map[string]string{
+						"allowed_methods.#": "1",
+						"allowed_origins.#": "1",
+					}),
+				),
+			},
+			{
+				Config: testAccSpacesBucketCORSConfigurationConfig_basic(rName, region, "https://www.example.com"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanSpacesBucketCorsConfigurationExists(ctx, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "cors_rule.#", "1"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "cors_rule.*", map[string]string{
+						"allowed_methods.#": "1",
+						"allowed_origins.#": "1",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccGetS3CorsConfigurationConn() (*s3.S3, error) {
+	client, err := acceptance.TestAccProvider.Meta().(*config.CombinedConfig).SpacesClient(testAccDigitalOceanSpacesBucketCorsConfiguration_TestRegion)
+	if err != nil {
+		return nil, err
+	}
+
+	s3conn := s3.New(client)
+
+	return s3conn, nil
+}
+
+func testAccCheckDigitalOceanSpacesBucketCorsConfigurationExists(ctx context.Context, resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Resource (%s) ID not set", resourceName)
+		}
+
+		s3conn, err := testAccGetS3CorsConfigurationConn()
+		if err != nil {
+			return err
+		}
+
+		response, err := s3conn.GetBucketCorsWithContext(context.Background(),
+			&s3.GetBucketCorsInput{
+				Bucket: aws.String(rs.Primary.Attributes["bucket"]),
+			})
+		if err != nil {
+			return fmt.Errorf("S3Bucket CORs error: %s", err)
+		}
+
+		if len(response.CORSRules) == 0 {
+			return fmt.Errorf("S3 Bucket CORS configuration (%s) not found", rs.Primary.ID)
+		}
+		return nil
+	}
+}
+
+func testAccCheckDigitalOceanSpacesBucketCorsConfigurationDestroy(s *terraform.State) error {
+	s3conn, err := testAccGetS3CorsConfigurationConn()
+	if err != nil {
+		return err
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		switch rs.Type {
+		case "digitalocean_spaces_bucket_cors_configuration":
+			_, err := s3conn.GetBucketCorsWithContext(context.Background(), &s3.GetBucketCorsInput{
+				Bucket: aws.String(rs.Primary.Attributes["bucket"]),
+			})
+			if err == nil {
+				return fmt.Errorf("Spaces Bucket Cors Configuration still exists: %s", rs.Primary.ID)
+			}
+
+		case "digitalocean_spaces_bucket":
+			_, err = s3conn.HeadBucket(&s3.HeadBucketInput{
+				Bucket: aws.String(rs.Primary.ID),
+			})
+			if err == nil {
+				return fmt.Errorf("Spaces Bucket still exists: %s", rs.Primary.ID)
+			}
+
+		default:
+			continue
+		}
+	}
+
+	return nil
+}
+
+func testAccSpacesBucketCORSConfigurationConfig_basic(rName string, region string, origin string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "foobar" {
+  name   = %s
+  region = %s
```

Comment:
```suggestion
  name   = "%s"
  region = "%s"
```

Think these need to be quoted or Terraform thinks it is meant to be a reference. 

```
    resource_spaces_bucket_cors_configuration_test.go:27: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid reference
        
          on terraform_plugin_test.tf line 3, in resource "digitalocean_spaces_bucket" "foobar":
           3:   name   = tf-acc-test-91z9i0ub6y
        
        A reference to a resource type must be followed by at least one attribute
        access, specifying the resource name.
        
        Error: Invalid reference
        
          on terraform_plugin_test.tf line 4, in resource "digitalocean_spaces_bucket" "foobar":
           4:   region = nyc3
        
        A reference to a resource type must be followed by at least one attribute
        access, specifying the resource name.
```